internal/setup/setup_test.go

```diff
@@ -851,3 +852,78 @@ func (s *S) TestParseRelease(c *C) {
 		}
 	}
 }
+
+// Test YAML marshalling of setup.Package and setup.Slice
+func (s *S) TestPackageMarshalYAML(c *C) {
+	sampleRelease := map[string]string{
+		"chisel.yaml": string(defaultChiselYaml),
+		"slices/mypkg.yaml": `
+			package: mypkg
+
+			slices:
+				foo:
+					contents:
+						/etc/foo:
+						/etc/foo-dir/: {make: true, mode: 0644}
+				bar:
+					essential:
+						- mypkg_foo
+					contents:
+						/bin/bar:
+						/etc/bar.conf: {text: TODO, mutable: true, arch: riscv64}
+						/lib/*-linux-*/bar.so: {arch: [amd64,arm64,i386]}
+						/usr/bin/bar: {symlink: /bin/bar}
+						/usr/bin/baz: {copy: /bin/bar}
+						/usr/lib/bar*.so:
+						/usr/share/bar/*.conf: {until: mutate}
+					mutate: |
+						dir = "/usr/share/bar/"
+						conf = [content.read(dir + path) for path in content.list(dir)]
+						content.write("/etc/bar.conf", "".join(conf))
+				baz:
+					essential:
+						- mypkg_foo
+						- mypkg_bar
+		`,
+	}
+
+	// write and parse the sample release
+	dir := c.MkDir()
+	for path, data := range sampleRelease {
+		fpath := filepath.Join(dir, path)
+		err := os.MkdirAll(filepath.Dir(fpath), 0755)
+		c.Assert(err, IsNil)
+		err = os.WriteFile(fpath, testutil.Reindent(data), 0644)
+		c.Assert(err, IsNil)
+	}
+	release, err := setup.ReadRelease(dir)
+	c.Assert(err, IsNil)
+	c.Assert(release, NotNil)
+	c.Assert(release.Path, Equals, dir)
+	release.Path = ""
+
+	// for each package, modify the sample release with marshalled yaml
+	for _, pkg := range release.Packages {
+		data, err := yaml.Marshal(pkg)
+		c.Assert(err, IsNil)
+		sampleRelease[pkg.Path] = string(data)
+	}
+
+	// write and parse the changed release
+	newDir := c.MkDir()
+	for path, data := range sampleRelease {
+		fpath := filepath.Join(newDir, path)
+		err := os.MkdirAll(filepath.Dir(fpath), 0755)
+		c.Assert(err, IsNil)
+		err = os.WriteFile(fpath, testutil.Reindent(data), 0644)
+		c.Assert(err, IsNil)
+	}
```

Comment:
I see that we are mutating `sampleRelease` so that we do not have to create `chisel.yaml` manually and we can create all the files in one loop. However it is not evident at all that this is the reason, can we include a comment or simplify the logic so that it is self-explanatory?

In my opinion, we could simplify it by creating the file outside of the loop but I am sure there are other ways.
```go
	newDir := c.MkDir()
	for _, pkg := range release.Packages {
		data, err := yaml.Marshal(pkg)
		c.Assert(err, IsNil)
		fpath := filepath.Join(newDir, pkg.Path)
		err = os.MkdirAll(filepath.Dir(fpath), 0755)
		c.Assert(err, IsNil)
		err = os.WriteFile(fpath, testutil.Reindent(string(data)), 0644)
		c.Assert(err, IsNil)
	}
	err = os.WriteFile(filepath.Join(newDir, "chisel.yaml"), testutil.Reindent(sampleRelease["chisel.yaml"]), 0644)
	c.Assert(err, IsNil)

```

Reply:
Nice, thanks. Seems like a better way to do this.